tools/ci/website_build.sh

```diff
@@ -0,0 +1,61 @@
+#!/bin/bash
+
+set -x
+
+neutral_status=78
+source_revision=$(git rev-parse HEAD)
+
+function modifies_relevant_files {
+  base_revision=$(
+    cat ${GITHUB_EVENT_PATH} | \
+      python -c 'import json, sys; print json.load(sys.stdin)["before"]'
+  )
+
+  git diff --name-only ${base_revision} | \
+    grep -E --silent '^(docs|tools)/'
+}
+
+if ! modifies_relevant_files ; then
+  echo No files related to the website have been modified. Exiting without
+  echo building.
+
+  exit ${neutral_status}
+fi
+
+git config --global user.email "wpt-pr-bot@users.noreply.github.com"
+git config --global user.name "wpt-pr-bot"
+
+cd docs
+
+pip install -r requirements.txt
+
+make html
+
+cd _build/html
+
+git init
```

Comment:
It seems nicer to do a shallow clone of the `gh_pages` branch and add the new content there so we get real history of the deployed content rather than overwriting it every time. Is there a reason not to do that?

Reply:
Only to avoid the additional complexity in this script. Having a meaningful commit history easily outweighs a few extra statements, though. Plus, this gives us the opportunity to communicate the condition where a patch that satisfies our criteria doesn't alter the built assets (e.g. changes to files in `tools/taskcluster/`).